### PR TITLE
Reorganizes the prizes in the Mining Vendor

### PR DESCRIPTION
**Pull request by PidgeyThePirate**
## What Does This PR Do
Reorganizes the mining vendor prizes into categories with more used categories at the top. The prizes in each category are alphabetically sorted. The default windows are a little taller.

## Why It's Good For The Game
Sorting through an unordered list to find your gear sucks even if you're used to it.

## Images of changes
image
image
image

## Changelog
:cl:
tweak: reorganized the mining vendor into categories and made the default window a little taller.
/:cl:

### CODE REVIEW

**Comment by Aidenzack1** [comment]
As a person who plays miner semi-frequently, I cant say how much I would love this simple quality of life improvement, and I cant tell you how much time I've wasted looking for something in the menu. The only thing more I could ask for would be a searchbar, but I could see the many problems with that

**Comment by Fox-McCloud** [comment]
This was rejected once before: https://github.com/ParadiseSS13/Paradise/pull/12451

This really needs to be made into a TGUI thing (once that gets merged), and the datums can be organized by category.

Abusing the fact that something doesn't have a price to be a breaker doesn't set well with me.

@Crazylemon64 might want to give your input.

**Comment by PidgeyThePirate** [comment]
It was rejected once before because it was packaged with a mining borg update and a new cell, and this part of it was considerably better received than either. TGUI isn't in yet and there's no guarantee that it will come in the immediate future. This will do fine as a stopgap, I think.

Still, this was a couple hours' work so I'm not married to it.

**Comment by AffectedArc07** [comment]
There should be lists for each category with categorisation headers as part of the UI itself. Making it so items with 0 cost dont show up is one of the hackiest things I have seen, and setting an item to 0 cost just to make it a linebreak is not a solution.

File: code/modules/mining/machine_vending.dm [hunk 1]
```diff
@@ -172,7 +209,7 @@
 	if(href_list["purchase"])
 		if(istype(inserted_id))
 			var/datum/data/mining_equipment/prize = locate(href_list["purchase"])
-			if(!prize || !(prize in prize_list) || prize.cost > inserted_id.mining_points)
+			if(!prize || prize.cost > inserted_id.mining_points)
```
Comment (farie82) [inline code/modules/mining/machine_vending.dm:212]: This is exploitable atm. Very minor exploit but.
In theory people can give a reference to a prize that is not part of this machine. For instance a grey slime extract from a golem vendor. If they somehow get that reference.

**Comment by PidgeyThePirate** [comment]
Oh, that one's up now.

**Pull request closed without merging**